File: 03-backend/app/schemas/terminal.py
from typing import List, Optional
from pydantic import BaseModel

class TerminalExecuteRequest(BaseModel):
    command: str
    args: Optional[List[str]] = None

class TerminalResultResponse(BaseModel):
    exit_code: int
    stdout: str
    stderr: str
    execution_time_ms: Optional[float] = None

File: frontend/src/types/terminal.ts
export interface TerminalExecutePayload {
  command: string;
  args?: string[];
}

export interface TerminalResult {
  exit_code: number;
  stdout: string;
  stderr: string;
  execution_time_ms?: number;
}

export interface TerminalHistoryItem {
  id: string;
  command: string;
  exit_code: number;
  created_at: string;
}
